## 43% of end-end tests have passed
```
Total executed tests: 7
Failed tests: 3
Expected to fail tests: 1
Passed tests: 3
```
```
IDs of failed tests: 727230,703133,727229 
```
```
IDs of expected to fail tests: 727232 
```
<h3>Failed tests (3)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=20479-skip-from-page-and-remove-account-switcher&PR_ID=20773&APK_NAME=20773.apk&TR_CASE_IDS=727230,703133,727229">Rerun failed tests</a></li><br/><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3623726">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td>

```
Test setup failed: ../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/connectionpool.py:703: in urlopen
    httplib_response = self._make_request(
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/connectionpool.py:449: in _make_request
    six.raise_from(e, None)
<string>:3: in raise_from
    ???
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/connectionpool.py:444: in _make_request
    httplib_response = conn.getresponse()
/usr/lib/python3.10/http/client.py:1375: in getresponse
    response.begin()
/usr/lib/python3.10/http/client.py:318: in begin
    version, status, reason = self._read_status()
/usr/lib/python3.10/http/client.py:287: in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
E   http.client.RemoteDisconnected: Remote end closed connection without response

During handling of the above exception, another exception occurred:
critical/test_wallet.py:23: in prepare_devices
    self.drivers, self.loop = create_shared_drivers(2)
base_test_case.py:309: in create_shared_drivers
    drivers = loop.run_until_complete(start_threads(test_suite_data.current_test.name,
/usr/lib/python3.10/asyncio/base_events.py:649: in run_until_complete
    return future.result()
__init__.py:26: in start_threads
    returns[k] = await returns[k]
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/appium/webdriver/webdriver.py:257: in __init__
    super().__init__(
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:206: in __init__
    self.start_session(capabilities)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/appium/webdriver/webdriver.py:346: in start_session
    response = self.execute(RemoteCommand.NEW_SESSION, w3c_caps)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:343: in execute
    response = self.command_executor.execute(driver_command, params)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/remote_connection.py:291: in execute
    return self._request(command_info[0], url, body=data)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/remote_connection.py:312: in _request
    response = self._conn.request(method, url, body=body, headers=headers)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/request.py:78: in request
    return self.request_encode_body(
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/request.py:170: in request_encode_body
    return self.urlopen(method, url, **extra_kw)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/poolmanager.py:376: in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/connectionpool.py:787: in urlopen
    retries = retries.increment(
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/util/retry.py:550: in increment
    raise six.reraise(type(error), error, _stacktrace)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/packages/six.py:769: in reraise
    raise value.with_traceback(tb)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/connectionpool.py:703: in urlopen
    httplib_response = self._make_request(
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/connectionpool.py:449: in _make_request
    six.raise_from(e, None)
<string>:3: in raise_from
    ???
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/connectionpool.py:444: in _make_request
    httplib_response = conn.getresponse()
/usr/lib/python3.10/http/client.py:1375: in getresponse
    response.begin()
/usr/lib/python3.10/http/client.py:318: in begin
    version, status, reason = self._read_status()
/usr/lib/python3.10/http/client.py:287: in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
 ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
```

<br/><br/></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3623725">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td>

```
Test setup failed: ../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/connectionpool.py:703: in urlopen
    httplib_response = self._make_request(
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/connectionpool.py:449: in _make_request
    six.raise_from(e, None)
<string>:3: in raise_from
    ???
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/connectionpool.py:444: in _make_request
    httplib_response = conn.getresponse()
/usr/lib/python3.10/http/client.py:1375: in getresponse
    response.begin()
/usr/lib/python3.10/http/client.py:318: in begin
    version, status, reason = self._read_status()
/usr/lib/python3.10/http/client.py:287: in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
E   http.client.RemoteDisconnected: Remote end closed connection without response

During handling of the above exception, another exception occurred:
critical/test_wallet.py:23: in prepare_devices
    self.drivers, self.loop = create_shared_drivers(2)
base_test_case.py:309: in create_shared_drivers
    drivers = loop.run_until_complete(start_threads(test_suite_data.current_test.name,
/usr/lib/python3.10/asyncio/base_events.py:649: in run_until_complete
    return future.result()
__init__.py:26: in start_threads
    returns[k] = await returns[k]
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/appium/webdriver/webdriver.py:257: in __init__
    super().__init__(
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:206: in __init__
    self.start_session(capabilities)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/appium/webdriver/webdriver.py:346: in start_session
    response = self.execute(RemoteCommand.NEW_SESSION, w3c_caps)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:343: in execute
    response = self.command_executor.execute(driver_command, params)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/remote_connection.py:291: in execute
    return self._request(command_info[0], url, body=data)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/remote_connection.py:312: in _request
    response = self._conn.request(method, url, body=body, headers=headers)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/request.py:78: in request
    return self.request_encode_body(
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/request.py:170: in request_encode_body
    return self.urlopen(method, url, **extra_kw)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/poolmanager.py:376: in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/connectionpool.py:787: in urlopen
    retries = retries.increment(
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/util/retry.py:550: in increment
    raise six.reraise(type(error), error, _stacktrace)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/packages/six.py:769: in reraise
    raise value.with_traceback(tb)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/connectionpool.py:703: in urlopen
    httplib_response = self._make_request(
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/connectionpool.py:449: in _make_request
    six.raise_from(e, None)
<string>:3: in raise_from
    ???
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/urllib3/connectionpool.py:444: in _make_request
    httplib_response = conn.getresponse()
/usr/lib/python3.10/http/client.py:1375: in getresponse
    response.begin()
/usr/lib/python3.10/http/client.py:318: in begin
    version, status, reason = self._read_status()
/usr/lib/python3.10/http/client.py:287: in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
 ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
```

<br/><br/></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3623724">test_restore_multiaccount_with_waku_backup_remove_switch</a>, id: 703133 </b></td></tr><tr><td><p><blockquote><div>## Signed in successfully!</div><div>Device 1: Find `CommunitiesTab` by `accessibility id`: `communities-stack-tab` </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:222: in test_restore_multiaccount_with_waku_backup_remove_switch
    self.home.communities_tab.click()
../views/base_element.py:90: in click
    element = self.find_element()
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 1: CommunitiesTab by accessibility id: `communities-stack-tab` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/80fcdfc85ffb463b863bacd79066523f?auth=f85b5d37762c83793b3b699ad8a18122#79">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Expected to fail tests (1)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3623728">test_wallet_add_remove_watch_only_account</a>, id: 727232 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `EditBox` by `accessibility id`: `add-address-to-watch` </div><div>Device 1: Type `0x8d2413447ff297d30bdc475f6d5cb00254685aae` to `EditBox` </div></blockquote></p>

```
critical/test_wallet.py:239: in test_wallet_add_remove_watch_only_account
    self.wallet_view.add_watch_only_account(address=address_to_watch, account_name=new_account_name)
../views/wallet_view.py:150: in add_watch_only_account
    self.account_has_activity_label.wait_for_visibility_of_element()
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 1: Text by accessibility id:`account-has-activity` is not found on the screen after wait_for_visibility_of_element 
```

<b>[[Missing networks in account address, https://github.com/status-im/status-mobile/issues/20166]]</b><br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a28562a3dd7444edbec5d360b8a6722a?auth=2b68ad7d458cef548d6c92b63a5845c1#100">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Passed tests (3)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3623727">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a28562a3dd7444edbec5d360b8a6722a?auth=2b68ad7d458cef548d6c92b63a5845c1#36">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3623722">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/80fcdfc85ffb463b863bacd79066523f?auth=f85b5d37762c83793b3b699ad8a18122#50">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3623723">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/c221caa1563b44c18abd4488c4561f64?auth=554e4801a224d059bd3658faec9d87af#97">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2309ebaf35c04fa4a9dbf80fa62cd3e2?auth=42cf1bb51b737afa2237c8fe7106263a#82">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

--- a/src/status_im/contexts/communities/actions/addresses_for_permissions/view.cljs
+++ b/src/status_im/contexts/communities/actions/addresses_for_permissions/view.cljs
@@ -262,7 +262,7 @@
 
         can-edit-addresses? (rf/sub [:communities/can-edit-shared-addresses? id])
 
-        wallet-accounts (rf/sub [:wallet/operable-accounts-without-watched-accounts])
+        wallet-accounts (rf/sub [:wallet/operable-accounts])
         joined (rf/sub [:communities/community-joined id])
         unmodified-addresses-to-reveal (rf/sub [:communities/addresses-to-reveal id])
         [addresses-to-reveal set-addresses-to-reveal] (rn/use-state unmodified-addresses-to-reveal)

--- a/src/status_im/contexts/wallet/common/token_value/view.cljs
+++ b/src/status_im/contexts/wallet/common/token_value/view.cljs
@@ -16,14 +16,14 @@
    :right-icon          :i/external})
 
 (defn- action-send
-  [send-params]
+  [send-params entry-point]
   {:icon                :i/send
    :accessibility-label :send
    :label               (i18n/label :t/send)
    :on-press            (fn []
                           (rf/dispatch [:hide-bottom-sheet])
                           (rf/dispatch [:wallet/clean-send-data])
-                          (rf/dispatch [:wallet/set-token-to-send send-params]))})
+                          (rf/dispatch [:wallet/set-token-to-send send-params entry-point]))})
 
 (defn- action-receive
   [selected-account?]
@@ -66,36 +66,41 @@
    :on-press            #(js/alert "to be implemented")})
 
 (defn token-value-drawer
-  [token watch-only?]
+  [token watch-only? entry-point]
   (let [token-symbol          (:token token)
         token-data            (first (rf/sub [:wallet/current-viewing-account-tokens-filtered
                                               token-symbol]))
         selected-account?     (rf/sub [:wallet/current-viewing-account-address])
+        token-owners          (rf/sub [:wallet/operable-addresses-with-token-symbol token-symbol])
         send-or-bridge-params (if selected-account?
                                 {:token       token-data
                                  :stack-id    :screen/wallet.accounts
-                                 :start-flow? true}
+                                 :start-flow? true
+                                 :owners      token-owners}
                                 {:token-symbol token-symbol
                                  :stack-id     :wallet-stack
-                                 :start-flow?  true})]
+                                 :start-flow?  true
+                                 :owners       token-owners})]
     [quo/action-drawer
      [(cond->> [(when (ff/enabled? ::ff/wallet.assets-modal-manage-tokens)
                   (action-manage-tokens watch-only?))
                 (when (ff/enabled? ::ff/wallet.assets-modal-hide)
                   (action-hide))]
         (not watch-only?) (concat [(action-buy)
-                                   (action-send send-or-bridge-params)
+                                   (when (seq token-owners)
+                                     (action-send send-or-bridge-params entry-point))
                                    (action-receive selected-account?)
                                    (when (ff/enabled? ::ff/wallet.swap) (action-swap))
-                                   (action-bridge send-or-bridge-params)]))]]))
+                                   (when (seq (seq token-owners))
+                                     (action-bridge send-or-bridge-params))]))]]))
 
 (defn view
-  [item _ _ {:keys [watch-only?]}]
+  [item _ _ {:keys [watch-only? entry-point]}]
   [quo/token-value
    (cond-> item
      (or (not watch-only?) (ff/enabled? ::ff/wallet.long-press-watch-only-asset))
      (assoc :on-long-press
             #(rf/dispatch
               [:show-bottom-sheet
-               {:content       (fn [] [token-value-drawer item watch-only?])
+               {:content       (fn [] [token-value-drawer item watch-only? entry-point])
                 :selected-item (fn [] [quo/token-value item])}])))])

--- a/src/status_im/contexts/wallet/events.cljs
+++ b/src/status_im/contexts/wallet/events.cljs
@@ -65,10 +65,21 @@
    {:db (assoc-in db [:wallet :current-viewing-account-address] address)}))
 
 (rf/reg-event-fx :wallet/clean-current-viewing-account
- (fn [{:keys [db]}]
-   (let [just-completed-transaction? (get-in db [:wallet :ui :send :just-completed-transaction?])]
-     (when-not just-completed-transaction?
-       {:db (update db :wallet dissoc :current-viewing-account-address)}))))
+ (fn [{:keys [db]} [ignore-just-completed-transaction?]]
+   (let [{:keys [entry-point just-completed-transaction?]} (-> db :wallet :ui :send)
+         entry-point-wallet-home?                          (= entry-point :wallet-stack)]
+     {:db (cond-> db
+            (and (not entry-point)
+                 (not ignore-just-completed-transaction?)
+                 (not just-completed-transaction?))
+            (update :wallet dissoc :current-viewing-account-address)
+
+            entry-point-wallet-home?
+            (update-in [:wallet :ui :send] dissoc :entry-point)
+
+            (and entry-point-wallet-home?
+                 (not just-completed-transaction?))
+            (update :wallet dissoc :current-viewing-account-address))})))
 
 (rf/reg-event-fx :wallet/close-account-page
  (fn [{:keys [db]}]

--- a/src/status_im/contexts/wallet/home/tabs/assets/view.cljs
+++ b/src/status_im/contexts/wallet/home/tabs/assets/view.cljs
@@ -18,4 +18,5 @@
       [rn/flat-list
        {:render-fn               token-value/view
         :data                    tokens
+        :render-data             {:entry-point :wallet-stack}
         :content-container-style style/list-container}])))

--- a/src/status_im/contexts/wallet/send/events.cljs
+++ b/src/status_im/contexts/wallet/send/events.cljs
@@ -193,25 +193,35 @@
 
 (rf/reg-event-fx
  :wallet/set-token-to-send
- (fn [{:keys [db]} [{:keys [token-symbol token stack-id start-flow?]}]]
+ (fn [{:keys [db]} [{:keys [token-symbol token stack-id start-flow? owners]} entry-point]]
    ;; `token` is a map extracted from the sender, but in the wallet home page we don't know the
    ;; sender yet, so we only provide the `token-symbol`, later in
    ;; `:wallet/select-from-account` the `token` key will be set.
-   (let [{token-networks :networks}                token
-         receiver-networks                         (get-in db [:wallet :ui :send :receiver-networks])
-         token-networks-ids                        (mapv #(:chain-id %) token-networks)
-         token-not-supported-in-receiver-networks? (not-any? (set receiver-networks)
-                                                             token-networks-ids)]
-     (when (or token token-symbol)
+   (let [{:keys [networks]}  token
+         receiver-networks   (get-in db [:wallet :ui :send :receiver-networks])
+         token-networks-ids  (map :chain-id networks)
+         unsupported-token?  (not-any? (set receiver-networks) token-networks-ids)
+         unique-owner        (when (= (count owners) 1)
+                               (first owners))
+         unique-owner-tokens (get-in db [:wallet :accounts unique-owner :tokens])
+         token-data          (or token
+                                 (when (and token-symbol unique-owner)
+                                   (some #(when (= (:symbol %) token-symbol) %)
+                                         unique-owner-tokens)))]
+     (when (or token-data token-symbol)
        {:db (cond-> db
-              :always      (update-in [:wallet :ui :send] dissoc :collectible)
-              :always      (assoc-in
-                            [:wallet :ui :send :token-not-supported-in-receiver-networks?]
-                            token-not-supported-in-receiver-networks?)
-              token        (assoc-in [:wallet :ui :send :token] token)
-              token        (assoc-in [:wallet :ui :send :token-display-name]
-                            (:symbol token))
-              token-symbol (assoc-in [:wallet :ui :send :token-symbol] token-symbol))
+              :always      (update-in [:wallet :ui :send]
+                                      #(-> %
+                                           (dissoc :collectible)
+                                           (assoc :token-not-supported-in-receiver-networks?
+                                                  unsupported-token?)))
+              token-symbol (assoc-in [:wallet :ui :send :token-symbol] token-symbol)
+              token-data   (update-in [:wallet :ui :send]
+                                      #(assoc %
+                                              :token              token-data
+                                              :token-display-name (:symbol token-data)))
+              unique-owner (assoc-in [:wallet :current-viewing-account-address] unique-owner)
+              entry-point  (assoc-in [:wallet :ui :send :entry-point] entry-point))
         :fx [[:dispatch [:wallet/clean-suggested-routes]]
              [:dispatch
               [:wallet/wizard-navigate-forward
@@ -225,9 +235,9 @@
    (let [{token-networks :networks
           token-symbol   :symbol}                  token
          receiver-networks                         (get-in db [:wallet :ui :send :receiver-networks])
-         token-networks-ids                        (mapv #(:chain-id %) token-networks)
-         token-not-supported-in-receiver-networks? (not (some (set receiver-networks)
-                                                              token-networks-ids))]
+         token-networks-ids                        (map :chain-id token-networks)
+         token-not-supported-in-receiver-networks? (not-any? (set receiver-networks)
+                                                             token-networks-ids)]
      {:db (-> db
               (assoc-in [:wallet :ui :send :token] token)
               (assoc-in [:wallet :ui :send :token-display-name] token-symbol)
@@ -256,7 +266,9 @@
 (rf/reg-event-fx
  :wallet/set-collectible-to-send
  (fn [{db :db} [{:keys [collectible current-screen start-flow?]}]]
-   (let [collection-data    (:collection-data collectible)
+   (let [viewing-account?   (some? (-> db :wallet :current-viewing-account-address))
+         entry-point        (when-not viewing-account? :wallet-stack)
+         collection-data    (:collection-data collectible)
          collectible-data   (:collectible-data collectible)
          contract-type      (:contract-type collectible)
          tx-type            (if (= contract-type constants/wallet-contract-type-erc-1155)
@@ -271,6 +283,7 @@
 
                               collectible
                               (str (:name collection-data) " #" collectible-id))
+         owner-address      (-> collectible :ownership first :address)
          collectible-tx     (-> db
                                 (update-in [:wallet :ui :send] dissoc :token)
                                 (assoc-in [:wallet :ui :send :collectible] collectible)
@@ -278,7 +291,14 @@
                                 (assoc-in [:wallet :ui :send :tx-type] tx-type))
          recipient-set?     (-> db :wallet :ui :send :recipient)]
      {:db (cond-> collectible-tx
-            one-collectible? (assoc-in [:wallet :ui :send :amount] 1))
+            :always
+            (assoc-in [:wallet :ui :send :entry-point] entry-point)
+
+            (not viewing-account?)
+            (assoc-in [:wallet :current-viewing-account-address] owner-address)
+
+            one-collectible?
+            (assoc-in [:wallet :ui :send :amount] 1))
       :fx [(when (and one-collectible? recipient-set?)
              [:dispatch [:wallet/get-suggested-routes {:amount 1}]])
            [:dispatch

--- a/src/status_im/contexts/wallet/send/from/view.cljs
+++ b/src/status_im/contexts/wallet/send/from/view.cljs
@@ -42,7 +42,6 @@
                                  {:on-press      #(rf/dispatch [:navigate-back])
                                   :margin-top    (safe-area/get-top)
                                   :switcher-type :select-account}]}
-
      [quo/page-top
       {:title                     (i18n/label :t/from-label)
        :title-accessibility-label :title-label}]

--- a/src/status_im/contexts/wallet/send/select_address/view.cljs
+++ b/src/status_im/contexts/wallet/send/select_address/view.cljs
@@ -177,7 +177,9 @@
                          (rf/dispatch [:wallet/clean-selected-collectible])
                          (rf/dispatch [:wallet/clean-send-address])
                          (rf/dispatch [:wallet/clean-disabled-from-networks])
-                         (rf/dispatch [:wallet/select-address-tab nil]))
+                         (rf/dispatch [:wallet/select-address-tab nil])
+                         (rf/dispatch [:wallet/clean-current-viewing-account
+                                       :ignore-just-complete-transaction]))
         on-change-tab  #(rf/dispatch [:wallet/select-address-tab %])
         input-value    (reagent/atom "")
         input-focused? (reagent/atom false)]

--- a/src/status_im/contexts/wallet/sheets/select_account/view.cljs
+++ b/src/status_im/contexts/wallet/sheets/select_account/view.cljs
@@ -20,7 +20,7 @@
 (defn view
   []
   (let [selected-account-address (rf/sub [:wallet/current-viewing-account-address])
-        accounts                 (rf/sub [:wallet/operable-accounts-without-watched-accounts])]
+        accounts                 (rf/sub [:wallet/operable-accounts])]
     [:<>
      [quo/drawer-top {:title (i18n/label :t/select-account)}]
      [gesture/flat-list

--- a/src/status_im/contexts/wallet/wallet_connect/session_proposal/view.cljs
+++ b/src/status_im/contexts/wallet/wallet_connect/session_proposal/view.cljs
@@ -70,7 +70,7 @@
 
 (defn- accounts-list
   []
-  (let [accounts         (rf/sub [:wallet/operable-accounts-without-watched-accounts])
+  (let [accounts         (rf/sub [:wallet/operable-accounts])
         selected-address (rf/sub [:wallet-connect/current-proposal-address])]
     [rn/view {:style style/account-switcher-list}
      (for [{:keys [address] :as account} accounts]

--- a/src/status_im/subs/community/account_selection.cljs
+++ b/src/status_im/subs/community/account_selection.cljs
@@ -53,7 +53,7 @@
 
 (re-frame/reg-sub :communities/accounts-to-reveal
  (fn [[_ community-id]]
-   [(re-frame/subscribe [:wallet/operable-accounts-without-watched-accounts])
+   [(re-frame/subscribe [:wallet/operable-accounts])
     (re-frame/subscribe [:communities/addresses-to-reveal community-id])])
  (fn [[accounts addresses] _]
    (filter #(contains? addresses (:address %))
@@ -61,7 +61,7 @@
 
 (re-frame/reg-sub :communities/airdrop-account
  (fn [[_ community-id]]
-   [(re-frame/subscribe [:wallet/operable-accounts-without-watched-accounts])
+   [(re-frame/subscribe [:wallet/operable-accounts])
     (re-frame/subscribe [:communities/airdrop-address community-id])])
  (fn [[accounts airdrop-address] _]
    (->> accounts

--- a/src/status_im/subs/wallet/wallet.cljs
+++ b/src/status_im/subs/wallet/wallet.cljs
@@ -134,10 +134,12 @@
                                                               (:chain-id %))))
                                      (map :chain-id)
                                      set)]
-     (assoc token
-            :networks          (network-utils/network-list token networks)
-            :available-balance (utils/calculate-total-token-balance token)
-            :total-balance     (utils/calculate-total-token-balance token enabled-from-chain-ids)))))
+     (some-> token
+             (assoc :networks          (network-utils/network-list token networks)
+                    :available-balance (utils/calculate-total-token-balance token)
+                    :total-balance     (utils/calculate-total-token-balance
+                                        token
+                                        enabled-from-chain-ids))))))
 
 (rf/reg-sub
  :wallet/wallet-send-token-symbol
@@ -464,22 +466,47 @@
  keep-operable-accounts)
 
 (rf/reg-sub
- :wallet/operable-accounts-without-watched-accounts
+ :wallet/operable-accounts
  :<- [:wallet/accounts-without-watched-accounts]
  keep-operable-accounts)
 
 (rf/reg-sub
+ :wallet/operable-addresses-tokens-with-positive-balance
+ :<- [:wallet/operable-accounts]
+ (fn [accounts]
+   (let [positive-balance-in-any-chain? (fn [{:keys [balances-per-chain]}]
+                                          (->> balances-per-chain
+                                               (map (comp :raw-balance val))
+                                               (some pos?)))]
+     (as-> accounts $
+       (group-by :address $)
+       (update-vals $ #(filter positive-balance-in-any-chain? (:tokens (first %))))))))
+
+(rf/reg-sub
  :wallet/accounts-with-current-asset
- :<- [:wallet/operable-accounts-without-watched-accounts]
+ :<- [:wallet/operable-accounts]
+ :<- [:wallet/operable-addresses-tokens-with-positive-balance]
  :<- [:wallet/wallet-send-token-symbol]
  :<- [:wallet/wallet-send-token]
- (fn [[accounts token-symbol token]]
-   (let [asset-symbol (or token-symbol (:symbol token))]
-     (if asset-symbol
-       (filter (fn [account]
-                 (some #(= (:symbol %) asset-symbol) (:tokens account)))
-               accounts)
-       accounts))))
+ (fn [[accounts addresses-tokens token-symbol token]]
+   (if-let [asset-symbol (or token-symbol (:symbol token))]
+     (let [addresses-with-asset (as-> addresses-tokens $
+                                  (update-vals $ #(set (map :symbol %)))
+                                  (keep (fn [[address token-symbols]]
+                                          (when (token-symbols asset-symbol) address))
+                                        $)
+                                  (set $))]
+       (filter #(addresses-with-asset (:address %)) accounts))
+     accounts)))
+
+(rf/reg-sub
+ :wallet/operable-addresses-with-token-symbol
+ :<- [:wallet/operable-addresses-tokens-with-positive-balance]
+ (fn [addresses-tokens [_ token-symbol]]
+   (keep (fn [[address tokens]]
+           (some #(when (= (:symbol %) token-symbol) address)
+                 tokens))
+         addresses-tokens)))
 
 (rf/reg-sub
  :wallet/account-tab

--- a/src/status_im/subs/wallet/wallet_test.cljs
+++ b/src/status_im/subs/wallet/wallet_test.cljs
@@ -13,16 +13,22 @@
               {:before #(reset! rf-db/app-db {})})
 
 (def ^:private accounts-with-tokens
-  {:0x1 {:tokens                    [{:symbol "ETH"} {:symbol "SNT"}]
+  {:0x1 {:tokens                    [{:symbol             "ETH"
+                                      :balances-per-chain {1 {:raw-balance "100"}}}
+                                     {:symbol             "SNT"
+                                      :balances-per-chain {1 {:raw-balance "100"}}}]
          :network-preferences-names #{}
          :customization-color       nil
          :operable?                 true
-         :operable                  :fully}
-   :0x2 {:tokens                    [{:symbol "SNT"}]
+         :operable                  :fully
+         :address                   "0x1"}
+   :0x2 {:tokens                    [{:symbol             "SNT"
+                                      :balances-per-chain {1 {:raw-balance "200"}}}]
          :network-preferences-names #{}
          :customization-color       nil
          :operable?                 true
-         :operable                  :partially}})
+         :operable                  :partially
+         :address                   "0x2"}})
 
 (def tokens-0x1
   [{:decimals                   1
@@ -495,11 +501,15 @@
            (assoc-in [:wallet :ui :send :token-symbol] "ETH")))
     (let [result (rf/sub [sub-name])]
       (is (match? result
-                  [{:tokens                    [{:symbol "ETH"} {:symbol "SNT"}]
+                  [{:tokens                    [{:symbol             "ETH"
+                                                 :balances-per-chain {1 {:raw-balance "100"}}}
+                                                {:symbol             "SNT"
+                                                 :balances-per-chain {1 {:raw-balance "100"}}}]
                     :network-preferences-names #{}
                     :customization-color       nil
                     :operable?                 true
-                    :operable                  :fully}]))))
+                    :operable                  :fully
+                    :address                   "0x1"}]))))
 
   (testing "returns the accounts list with the current asset using token"
     (swap! rf-db/app-db
@@ -508,11 +518,15 @@
            (assoc-in [:wallet :ui :send :token] {:symbol "ETH"})))
     (let [result (rf/sub [sub-name])]
       (is (match? result
-                  [{:tokens                    [{:symbol "ETH"} {:symbol "SNT"}]
+                  [{:tokens                    [{:symbol             "ETH"
+                                                 :balances-per-chain {1 {:raw-balance "100"}}}
+                                                {:symbol             "SNT"
+                                                 :balances-per-chain {1 {:raw-balance "100"}}}]
                     :network-preferences-names #{}
                     :customization-color       nil
                     :operable?                 true
-                    :operable                  :fully}]))))
+                    :operable                  :fully
+                    :address                   "0x1"}]))))
 
   (testing
     "returns the full accounts list with the current asset using token-symbol if each account has the asset"

